api/environments/serializers.py

```diff
@@ -97,6 +98,16 @@ def get_project(
             "Unable to retrieve project for metadata validation."
         )
 
+    def update(
+        self, instance: models.Model, validated_data: dict[str, typing.Any]
+    ) -> Environment:
+        metadata_items = validated_data.pop("metadata", [])
+        environment = super().update(instance, validated_data)  # type: ignore[no-untyped-call]
+        self.update_metadata(environment, metadata_items)
+        environment.refresh_from_db()
+        assert isinstance(environment, Environment)
```

Comment:
Is this **needed**? Not a blocker.

Reply:
Not needed per say but mypy is a tough opponent and it would require unrelated typing that I think is overkill in the context of a bug fix

Reply:
Fairy nuff. Let me know if you need help swinging a few punches.